public/diagnostic.js

```diff
@@ -23,3 +23,34 @@
 //  -  Order 1 : 2 hats ($5 each)
 //  -  Order 2 : 1 desk lamp ($20 each)
 //  -  Order 3 : 3 hand towels ($8 each)
+
+const Order = Ember.Object.extend({
+  unitPrice: NaN,
+  quantity: NaN,
+  name: '',
+  orderPrice: Ember.computed('unitPrice', 'quantity', function() {
+    return unitPrice * quantity;
+  }),
+});
+
+const Cart = Ember.Object.extend({
+  orders: [],
+  addToCart: function(order) {
+    this.orders.push(order);
+  },
+  totalPrice: Ember.computed('orders.@each', function() {
+    let hold = orders.map((val) => { return val.orderPrice; });
+      return hold.reduce((a,b) => a + b);
+  }),
+});
```

Comment:
orders needs to be assigned in `totalPrice`, like this, `let orders = this.get('orders')`.  Just as in `addToCart`, in order to take advantage of the benefits of using Ember you need to use getters and setters, so `val.get('orderPrice')` instead of `val.orderPrice`